### PR TITLE
Merge upsteam changes from elementary

### PR DESCRIPTION
**Pull request by mmstick**
It's been a while since we've synced with elementary. The changes are huge (99% po files). This will need some careful review. It seems to be working though.

screenshot from 2018-06-12 17-03-16
screenshot from 2018-06-12 17-03-09
screenshot from 2018-06-12 17-02-57
screenshot from 2018-06-12 17-02-49
screenshot from 2018-06-12 17-02-40
screenshot from 2018-06-12 17-02-35

### CODE REVIEW

**Comment by mmstick** [comment]
@cassidyjames

**Comment by mmstick** [comment]
I feel like elementary should separate their potfiles from their projects. It makes the commit stream way too noisy.

**Comment by cassidyjames** [comment]
@mmstick I think at a glance we lost a style change for the headerbar. I think master has a little less padding.

**Comment by cassidyjames** [comment]
Something @danrabbit has mentioned several times is carrying patches to upstream projects instead of complete forks. It tends to make the diffs much smaller, and it's what we do with big complex apps like Epiphany in elementary OS. I'm not sure if translations can live in an orphan branch due to how they're generated and updated through Weblate.

**Comment by brs17** [comment]
I notice that this pr removes the ISSUE_TEMPLATE.md that we have and want to keep. We will need to add that as well...commit incoming!

File: src/Widgets/Banner.vala [hunk 1]
```diff
@@ -22,12 +22,12 @@ const string BANNER_STYLE_CSS = """
...
     @define-color banner_fg_color %s;
 
-    .banner { 
+    .banner {
         transition: all %ums ease-in-out;
     }
 """;
 
-const string DEFAULT_BANNER_COLOR_PRIMARY = "#454241";
+const string DEFAULT_BANNER_COLOR_PRIMARY = "#68758e";
```
Comment (cassidyjames) [inline src/Widgets/Banner.vala]: We'll want to revert this change. The old color is Pop brown, this new color is the elementary slate.

**Comment by cassidyjames** [comment]
Overall the changes look sane; it's mostly code cleanup and then paid-app-related changes that don't affect us. The one neat user-facing thing I noticed is that this should get us better icons on HiDPI.

**Comment by brs17** [comment]
I can test that this builds/works in Pop!_OS, I imagine between @cassidyjames and @mmstick you two have looked at the files that this pr removes? (*CMakeLists.txt and config.vala.in)

**Comment by cassidyjames** [comment]
@brs17 yes, upstream changed from CMake to Meson a couple of months ago.

**Comment by brs17** [comment]
With @mmstick's latest commits I was able to build and run this pr. Things I noticed:

1. `Non-curated apps`, if we don't have curated apps, we shouldn't label everything "non curated".

screenshot from 2018-07-02 10-38-01

2. A "Share" button has been added. Are we ok with the share options provided?
screenshot from 2018-07-02 10-39-05

3. title-bar is massive. (see other screenshots). Compared with what we currently ship:
screenshot from 2018-07-02 10-42-02

4. Original icon and not our custom icon. Also in the application menu the app is labeled as App Center
screenshot from 2018-07-02 10-39-19

5. Repoman is not launched when selecting the icon. `apt` shows it as unnecessary and that it can be removed along with some python dependencies.

We definitely need a better process to deal with upstream merges. As @cassidyjames had mentioned, perhaps for reasons like this, we should just be carrying patches and not running a forked version?

**Comment by cassidyjames** [comment]
> Non-curated apps, if we don't have curated apps, we shouldn't label everything "non curated".

It looks like there is a `CURATED` flag: https://github.com/elementary/appcenter/pull/516/files though I'm not sure how to set this in Meson.

>A "Share" button has been added. Are we ok with the share options provided?

No, this doesn't make sense with Pop!_Shop since we don't run an appstream redirect site. We should re-remove it as we did before.

> title-bar is massive. (see other screenshots). Compared with what we currently ship:

Yeah, we removed some margins downstream before due to a difference in styling between Pop and elementary. We should re-remove the headerbar margins.

>Original icon and not our custom icon. Also in the application menu the app is labeled as App Center

Yeah we need to walk through and double-check all user-facing references to AppCenter. The icon ideally would be provided by the OS icon set, though, not the application itself.

>Repoman is not launched when selecting the icon. apt shows it as unnecessary and that it can be removed along with some python dependencies.

Probably just need to update the debian packaging to re-require repoman as a dependency.

**Comment by brs17** [comment]
@cassidyjames I'd like to get your eyes on it once more before merging.

**Comment by cassidyjames** [comment]
It looks like displaying updates may be broken in this branch. In elementary master, running `killall io.elementary.appcenter; io.elementary.appcenter --fake-package-update=firefox` properly loads in apps from the API, then checks for updates, and displays any available updates (including the fake Firefox one) on the Installed page (plus a badge on the tab with a number of updates). In this PR, I don't get any real or faked updates.

**Comment by cassidyjames** [comment]
Hm, our `master_bionic` branch also appears to be broken in the same way. So it might not be a regression in this branch specifically.

**Comment by brs17** [comment]
If the issue you found is happening in master regardless, I feel like we should go ahead and merge this pr as it is not a regression.

**Comment by mmstick** [comment]
I'll look into the other issues soon. Would like to get our code base closer to elementary's. Using conditional compilation seems the simplest way to co-exist side by side. Meson also simplifies a lot of things.

**Comment by cassidyjames** [comment]
Badge issue reported at https://github.com/pop-os/shop/issues/85
